### DISCUSSION TITLE
How to setup a teleport cluster with two proxies and two auth instances on separate servers

### DISCUSSION THREAD

**Discussion by johanwesterduin** [gravitational/teleport#9009] [Q&A]
Hi to all readers,

for our business we are setting up a teleport cluster with four servers (two proxy servers and two auth servers) based on the opensource version.

As far we can see all examples are based on a setup with one server without certificates and we miss the necessary information how to correctly setup an additional proxy server or how to setup en separate proxy and auth server.

Setting up a single server with the teleport proxy and auth service with certificates is easy, but add an extra proxy or separate these services we missing crucial information how to do that.

Does anyone have experience with a design with:
- two proxies?
- separate servers for proxy and auth server?

Does anyone have an example for these two questions?

Any help is welcome.
Thanks in advance.
Johan

**Comment by webvictim** [comment]
Proxies are stateless - you can run as many of them as you like and connect them to a load balancer which routes to multiple auth servers.

Auth servers have state and as such must use an HA backend if you want to have more than one. There is a section on storage backends in our docs here: https://goteleport.com/docs/setup/reference/backends

You may find that using `etcd` is useful to host multiple on-premise auth servers. You will also need a TCP load balancer to route traffic to multiple auth servers, or potentially a round-robin DNS record which points to all the auth server IPs.
